### PR TITLE
M3-3868 Change: Include LKE clusters in Support tickets dropdown

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

I went a little rogue on this one, sorry in advance.

The only actual change here (should be) that when creating a Support ticket, LKE clusters appear as one of the possible "What is this regarding?" options. However, when I tried to do this I noticed that this select was making a getAll() request _on every change_. This is old and ridiculous (I wrote it), and the straw that broke the camel's back on this component. I re-wrote it as a function component so that I could use hooks. Then (sorry) I wrote a bunch of hooks for the major entity types, and combined them into a `useEntities` hook that provides data and request handlers for each of the types. This way, the support ticket drawer only requests data if it doesn't already have it available in Redux.

- Rewrote the tests to use RTL
- Adjusted types

Bottom line: everything should be exactly the same as before, except that you can make a ticket about an LKE cluster. Please test everything about support ticket creation (using prefilled title and description, adding/removing attachments, selecting different entity types, etc.) and verify that there are no regressions. Please also try request blocking various entity types to see the error handling.

### CODE REVIEW

File: packages/manager/src/store/linodes/linodes.reducer.ts [hunk 1]
```diff
@@ -40,7 +40,8 @@ const reducer: Reducer<State> = (state = defaultState, action) => {
   if (isType(action, getLinodesActions.started)) {
     return {
       ...state,
-      loading: true
+      loading: true,
+      error: { ...state.error, read: undefined }
```
Comment (Jskobos) [inline packages/manager/src/store/linodes/linodes.reducer.ts]: I guess we've never relied on this before, but this is how the logic ought to have been from the beginning.

File: packages/manager/src/store/kubernetes/kubernetes.requests.ts [hunk 2]
```diff
@@ -32,11 +32,12 @@ export const requestKubernetesClusters: ThunkActionCreator<
       for (; i < data.length; i++) {
         dispatch(requestNodePoolsForCluster({ clusterID: data[i].id }));
       }
-      return dispatch(
+      dispatch(
         requestClustersActions.done({
           result: data
         })
       );
+      return data;
```
Comment (Jskobos) [inline packages/manager/src/store/kubernetes/kubernetes.requests.ts]: This follows the existing pattern for Thunks, so that consumers can do `thunkRequest().then(response => whatever)`

File: packages/manager/src/hooks/useEntities.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  const domains = _domains.data ?? [];
+  const images = (Object.values(_images.data) ?? []).filter(
+    thisImage => !thisImage.is_public
+  );
+  const volumes = Object.values(_volumes.itemsById);
+  const kubernetesClusters = _kubernetesClusters.entities;
+  const nodeBalancers = Object.values(_nodeBalancers.itemsById);
+
+  return {
+    domains: {
```
Comment (Jskobos) [inline packages/manager/src/hooks/useEntities.ts]: I kept adding more and more of these fields as I needed them, and at this point it's basically the entire Redux state for each entity. I still like this format but it's a little redundant.

File: packages/manager/src/hooks/useImages.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { Image } from 'linode-js-sdk/lib/images/types';
```
Comment (Jskobos) [inline packages/manager/src/hooks/useImages.ts]: These get repetitive but it's the best way I could think of to manage the types, location in the store, etc. for the different entities.

File: packages/manager/src/hooks/useEntities.ts (hunk 3 above)
Comment (acourdavault) [reply packages/manager/src/hooks/useEntities.ts]: Would there be a performance hit to passing the whole state?

While we do not want to do that everywhere, I feel like we are introducing risk and complexity by writing this whole big object to share the data

File: packages/manager/src/features/Support/SupportTickets/SupportTicketDrawer.test.tsx [hunk 5]
```diff
@@ -1,13 +1,11 @@
...
 import { getVersionString } from 'src/utilities/getVersionString';
-import {
-  CombinedProps,
-  State,
-  SupportTicketDrawer
-} from './SupportTicketDrawer';
+import { renderWithTheme } from 'src/utilities/testHelpers';
+import SupportTicketDrawer, { Props } from './SupportTicketDrawer';
+
+const support = require.requireMock('linode-js-sdk/lib/support');
```
[the remark this answers is not in the document]
Comment (Jskobos) [reply packages/manager/src/features/Support/SupportTickets/SupportTicketDrawer.test.tsx]: I've done this before and I'm not sure I understand it well enough to explain it. When you do the `jest.mock()` business, you can't refer to any functions inside it by name since they're out of scope. So you have to requireMock the module and replace the method you want to assert against with another mock. `requireMock` doesn't actually mock anything though, so the jest.mock() call is still necessary. I think.

File: packages/manager/src/hooks/useEntities.ts (hunk 3 above)
Comment (Jskobos) [reply packages/manager/src/hooks/useEntities.ts]: There wouldn't be a performance hit that I can think of. I only did this because we don't (yet) have a consistent state for all of these entity types in Redux, so someone somewhere has to do all the conversions in this file to get consistent data (e.g. we sometimes just return `entities` and sometimes `Object.keys(itemsById)` bc the Redux store shapes are inconsistent). Doing it here lets us export a consistent and typed interface that consumers can read without worrying about the quirks of our Redux store.
